zipkin-ui/js/component_ui/uploadTrace.js

```diff
@@ -0,0 +1,44 @@
+import {component} from 'flightjs';
+import FullPageSpinnerUI from '../component_ui/fullPageSpinner';
+import traceToMustache from '../../js/component_ui/traceToMustache';
+
+export default component(function uploadTrace() {
+  this.doUpload = function() {
+    const files = this.node.files;
+    if (files.length === 0) {
+      return;
+    }
+
+    const reader = new FileReader();
+    reader.onload = evt => {
+      let model;
+      try {
+        const trace = JSON.parse(evt.target.result);
```

Comment:
I think here, we should assume zipkin v2 format unless we know heuristically it is not. Worst case we can copy/paste a modified version of https://github.com/openzipkin/zipkin-js/blob/master/packages/zipkin/src/jsonEncoder.js#L25 to map v2 json to v1 format.  Ex here are the tests for that: https://github.com/openzipkin/zipkin-js/blob/master/packages/zipkin/test/jsonEncoder.JSON_V1.test.js

Reason is that it is only time resource which is why the UI isn't internally rewritten to v2 yet. I don't want to encourage people to keep using v1. The best heuristic to detect if incoming json is v2 or not is presence of the "localEndpoint" field, which is only absent on instrumentation error.

Reply:
Are you suggesting that we iterate through the spans in `trace`, see if they have a 'localEndpoint', and if they do, then call `jsonEncoder.encodeV1()` (effectively) on each span so that the UI can display both v1 and v2 traces?

I'm midly afraid that v2 traces with 9k spans will suffer a bit from that.  But that concern aside, is there an easy way to extract jsonEncoder without having to copy/paste it into zipkin-ui?  Of course, I'm talking about extracting/sharing the conversion more so than the call to `JSON.stringify()` at the end.